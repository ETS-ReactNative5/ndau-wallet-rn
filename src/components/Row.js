import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Text, TouchableHighlight, Platform } from 'react-native';

class Row extends React.PureComponent {
  render() {
    const { title, onPress, onPressIn, platform, testID } = this.props;
    if (platform && platform !== Platform.OS) {
      return <View />;
    }

    return (
      <TouchableHighlight
        onPress={onPress}
        onPressIn={onPressIn}
        testID={testID}
        underlayColor={'rgba(255, 255, 255, 0.054)'}
      >
        <View style={styles.row}>
          <Text style={styles.text}>{title}</Text>
        </View>
      </TouchableHighlight>
    );
  }
}

Row.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  onPressIn: PropTypes.func
};

const styles = StyleSheet.create({
  row: {
    height: 68,
    paddingHorizontal: 16,
    paddingBottom: 20,
    paddingTop: 20,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(255, 255, 255, 0.054)'
  },
  text: {
    color: '#ffffff',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Regular'
  }
});

export default Row;
