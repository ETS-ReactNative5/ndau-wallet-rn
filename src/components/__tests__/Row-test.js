import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import Row from '../Row';

import renderer from 'react-test-renderer';

describe('testing Row...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Row parentStyles={styles} navigator={navigator} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
