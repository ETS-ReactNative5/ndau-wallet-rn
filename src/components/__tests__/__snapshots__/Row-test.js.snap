// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Row... renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "rgba(255, 255, 255, 0.054)",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 68,
        "justifyContent": "flex-start",
        "paddingBottom": 20,
        "paddingHorizontal": 16,
        "paddingTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;
